plugins/async-node/jvm/src/test/kotlin/com/intuit/playerui/plugins/asyncnode/AsyncNodePluginTest.kt

```diff
@@ -228,4 +232,162 @@ internal class AsyncNodePluginTest : PlayerTest() {
         }
         Assertions.assertTrue(true)
     }
+
+    @TestTemplate
+    fun `handle multiple updates through callback mechanism`() = runBlockingTest {
+        // TODO: This typing is not great - need to go fix hook types
+        var deferredResolve: ((asyncNodeUpdate) -> Unit)? = null
+        var updateContent: ((asyncNodeUpdate) -> Unit)? = null
+
+        var count = 0
+
+        plugin.hooks.onAsyncNode.tap("") { _, node, callback ->
+            updateContent = callback
+            val result = suspendCoroutine { cont ->
+                deferredResolve = { value ->
+                    cont.resume(value)
+                }
+            }
+            BailResult.Bail(result)
+        }
+
+        var viewUpdateContinuation: Continuation<Asset?>? = null
+        player.hooks.view.tap { v ->
+            v?.hooks?.onUpdate?.tap { asset ->
+                count++
+                viewUpdateContinuation?.resume(asset)
+                viewUpdateContinuation = null
+            }
+        }
+        player.start(asyncNodeFlowSimple)
+
+        var view = player.inProgressState?.lastViewUpdate
+        Assertions.assertNotNull(view)
+        Assertions.assertEquals(
+            "action",
+            view!!.getList("actions")?.filterIsInstance<Node>()?.get(0)?.getObject("asset")?.get("type"),
+        )
+        Assertions.assertEquals(1, view.getList("actions")?.size)
+
+        while (true) {
+            if (deferredResolve != null) {
+                break
+            }
+
+            yield()
+        }
+
+        // create a view object to pass it to the deferred resolve
+        val viewObject = mapOf(
+            "asset" to mapOf(
+                "id" to "asset-1",
+                "type" to "action",
+                "value" to "New asset!",
+            ),
+        )
+
+        deferredResolve!!.invoke(listOf(viewObject))
+
+        Assertions.assertEquals(1, count)
```

Comment:
I _think_ I'm following this, can you tell me why this is 1 and not 2 at this point

Reply:
> The assertion is placed after the first update event has occurred, so the count should be 1. If the count were 2, it would indicate that two update events have occurred,

Is this `deferredResolve` not triggering the second update? First update being the initial view without any async resolve?